### PR TITLE
Fix Android Quickstart issues [QS-412]

### PR DESCRIPTION
**Pull request by Widcket**
- Removed a section with repeated information (**Capture the Result**) and relocated the unique bits of information it contained.
- Fixed formatting details in the code snippets.
- Updated the OkHttp example snippet for the latest version of that library.
- Reworded a couple of sentences.

See a detailed breakdown of the issues in [this spreadsheet](https://docs.google.com/spreadsheets/d/1bWrIhLP29DXz_l5aps8icdILXYxw4EOQooYoWRxXZ5k/edit?usp=sharing).

The fixes in the sample apps can be found in [this PR](https://github.com/auth0-samples/auth0-android-sample/pull/82).

### CODE REVIEW

File: articles/quickstart/native/android/06-calling-apis.md [hunk 1]
```diff
@@ -102,16 +104,16 @@ Tell the client to create a new `Call` with the request you created. Call the `e
 
 client.newCall(request).enqueue(new Callback() {
     @Override
-    public void onFailure(Request request, IOException e) {
-        //show error
+    public void onFailure(Request request, final IOException e) {
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/06-calling-apis.md:107]: Any reason to change this to `final` here?

File: articles/quickstart/native/android/07-authorization.md [hunk 2]
```diff
@@ -21,40 +21,17 @@ This tutorial shows you how to use Auth0 to create access roles for your users.
 Be sure that you have completed the [Login](/quickstart/native/android/00-login) quickstart.
 :::
 
-Create a rule that assigns the users either an `admin` role, or a simple `user` role. Go to the [new rule page](${manage_url}/#/rules/new) and select the "Set Roles To A User" template, under **Access Control**. Replace the default script contents with the following snippet:
+Create a [Rule](https://auth0.com/docs/rules) that assigns the users either an `admin` role, or a simple `user` role. Go to the [new rule page](${manage_url}/#/rules/new) and select the **Set Roles To A User** template, under **Access Control**. Edit the following lines from the default script to match the conditions that fit your needs:
 
 ```js
-function (user, context, callback) {
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/07-authorization.md:27]: this line is the signature of a typical Auth0 Rule. If you remove it, how does the user know they need to put the code below "inside" the default rule code ?

File: articles/quickstart/native/android/_includes/_login.md [hunk 3]
```diff
@@ -20,94 +20,48 @@ private Auth0 auth0;
...
-Future steps of this quickstart guide move the `AuthCallback` in-line instance into a class field named `loginCallback`.
-:::
-
-### Capture the Result
+            .withScheme("demo")
+            .withAudience(String.format("https://%s/userinfo", getString(R.string.com_auth0_domain)))
+            .start(MainActivity.this, new AuthCallback() {
+                @Override
+                public void onFailure(@NonNull Dialog dialog) {
+                // Show error Dialog to user
```
Comment (lbalmaceda) [inline articles/quickstart/native/android/_includes/_login.md]: Indentation of the comments (here and below)

Comment (Widcket) [reply articles/quickstart/native/android/_includes/_login.md]: Fixed in 8e7e072b345e65b094a662b6e0c95dcd502ebd7f.

File: articles/quickstart/native/android/07-authorization.md (hunk 2 above)
Comment (Widcket) [reply articles/quickstart/native/android/07-authorization.md:27]: The user does not need to put the code inside the Rule, they just have to search for the relevant lines and change them as needed. Hence the (new) instruction above reads: `Edit the following lines from the default script to match the conditions that fit your needs`.

File: articles/quickstart/native/android/06-calling-apis.md (hunk 1 above)
Comment (Widcket) [reply articles/quickstart/native/android/06-calling-apis.md:107]: If I remember correctly, that's how Android Studio generates the callback implementation. I can change it back and remove the `final`, what do you think?

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**